LoopKit/GlucoseRangeSchedule.swift

```diff
@@ -57,8 +57,45 @@ extension DoubleRange: Hashable {}
 public struct GlucoseRangeSchedule: DailySchedule, Equatable {
     public typealias RawValue = [String: Any]
 
+    /// A time-based value overriding the rangeSchedule
+    public struct Override: Equatable {
+
+        public let start: Date
+        public let end: Date
+        public let value: DoubleRange
+
+        /// Initializes a new override
+        ///
+        /// - Parameters:
+        ///   - start: The date at which the override starts
+        ///   - end: The date at which the override ends, or nil for an indefinite override
+        ///   - value: The value to return when active
+        public init(start: Date, end: Date?, value: DoubleRange) {
+            self.start = start
+            self.end = end ?? .distantFuture
+            self.value = value
+        }
+
+        public var activeDates: DateInterval {
+            return DateInterval(start: start, end: end)
+        }
+
+        public func isActive(at date: Date = Date()) -> Bool {
+            return activeDates.contains(date) && !value.isZero
```

Comment:
Why does a glucose range of 0 mean the schedule is not active? Is it expected that users enters 0 instead of defining times/dates for when the schedule is active? Also, if the value is instead a negative number or a small value that isn't considered 0, could the schedule be considered active?

Reply:
This was to handle previously stored ranges of 0-0, which were indicated no range set. Now, when you create an override without a range adjustment, it is null, and this object is not created.  The check does include values < .ulpOfOne, if you look at the definition of isZero.

We don't have guardrails when setting target ranges yet, and Overrides are currently not slated to be included in 1.0 of Tidepool Loop. We have a feature flag that turns them off in the Tidepool Loop build.